ref-tests/tests/ic-ref.rs

```diff
@@ -770,3 +770,124 @@ mod extras {
         })
     }
 }
+
```

Comment:
The tests in this file is meant to more or less mirror this Spec tests from this interface spec file: https://github.com/dfinity/ic-hs/blob/master/src/IC/Test/Spec.hs
Please move them to https://github.com/dfinity/agent-rs/blob/main/ref-tests/tests/integration.rs